enhancements/agent-installer/agent-based-installer.md

```diff
@@ -0,0 +1,279 @@
+
+Each of the current installation options has limitations that make it
+unpalatable to a subset of users. IPI is highly opinionated, and suffers from
+providing limited visibility of configuration problems in on-premises
+environments. UPI exposes users to the full complexity of OpenShift to the user
+with little guidance. The (hosted) Assisted service is not available for
+IPv6-only or disconnected environments. For some users, none of these options
+is optimal.
+
+On-premises users need a simple way to deploy their first cluster, even in a
+disconnected environment, using their own automation to boot hosts.
+
+### Goals
+
+- Install clusters in fully-disconnected environments.
+- Make use of assisted installation components to pre-validate the installation
+  environment.
+- Require no machines in the cluster environment other than those that are to
+  become part of the cluster.
+- Install clusters with single-node, compact, and high-availability topologies.
+- Be agnostic to the tools used to provision (virtual) machines.
+- Support install configurations with platform `none`, `baremetal`, or
+  `vsphere`.
+
+### Non-Goals
+
+- Replace the IPI installer in any capacity.
+- Automate booting of the image on (virtual) machines.
+- Support install configurations for cloud-based platforms.
+- Provide an interactive API to drive automated deployments.
+- Install minor versions of OpenShift that are older than the agent-based
+  installer version.
+- Operate in disconnected environments without the need to separately mirror
+  the release payload to a registry that is usable from the installed cluster.
+
+## Proposal
+
+A command-line tool will enable users to build a custom RHCOS image, initially
+in ISO format, containing the components needed to install. The base ISO will
+be [retrieved](https://github.com/openshift/machine-os-images/) from the
+release payload. Users can then boot this image on the hosts that are to become
+part of the cluster using any method they choose - this could be existing
+automation that they have or manually booting using physical media. One host
+(hereafter referred to as "Node 0") will be selected to run the
+assisted-service. All hosts will run the assisted agent. A single ISO will be
+used for all hosts in the cluster. Required container images will be pulled
+from the payload of the OpenShift release that is being installed.
+
+In the automated workflow, all configuration data for the cluster to be
+installed (including the expected number of hosts) will be provided at the time
+the ISO is generated. Installation will proceed automatically once the
+requisite hosts are available and have passed validations. An API running on
+Node 0 will enable automated tools to monitor progress, detect errors, and
+retrieve logs until the OpenShift control plane is available.
+
+In the interactive flow, the ISO will be generated with only enough data to
```

Comment:
I still have questions about whether what's described here is the best UX we can provide. Let's put the interactive flow on the non-goals list for this document and focus on the automated flow.